### PR TITLE
[Broker] Make PersistentTopicsBase#internalCreateNonPartitionedTopic async.

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

Make PersistentTopicsBase#internalCreateNonPartitionedTopic async.

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

**Review by codelipenghui — DISMISSED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -372,37 +372,42 @@ protected void internalRevokePermissionsOnTopic(String role) {
...
+                   NamespaceOperation.CREATE_TOPIC))
+           .thenCompose(__ -> getPartitionedTopicMetadataAsync(topicName, false, false))
+           .thenAccept(partitionMetadata -> {
+               if (partitionMetadata.partitions > 0) {
+                   log.warn("[{}] Partitioned topic with the same name already exists {}", clientAppId(), topicName);
+                   throw new RestException(Status.CONFLICT, "This topic already exists");
+               }
+           })
+           .thenCompose(__ -> pulsar().getBrokerService().getTopicIfExists(topicName.toString()))
+           .thenAccept(existedTopic -> {
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:392]: maybe we can use one ``thenAccept`` to handle with 398

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -372,37 +372,42 @@ protected void internalRevokePermissionsOnTopic(String role) {
...
+               if (existedTopic.isPresent()) {
+                   log.error("[{}] Topic {} already exists", clientAppId(), topicName);
+                   throw new RestException(Status.CONFLICT, "This topic already exists");
+               }
+           })
+           .thenAccept(__ ->
+               pulsar().getBrokerService().getTopic(topicName.toString(), true, properties)
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:399]: Ummm, are you missing return success?

**Review by mattisonchao — COMMENTED** [review]
Another, I have two suggestions:

- Remove asyncResponse from the internal method, and unified exception and result handling are performed at the controller layer, which makes the internal method purer.
- Have you checked to see if there is a test for the modified method?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 2 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:399]: Yes, fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:392]: Yes, good suggestion.

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:392]: fixed. thanks.

**Comment by Technoboy-** [comment]
> Another, I have two suggestions:
> 
> * Remove asyncResponse from the internal method, and unified exception and result handling are performed at the controller layer, which makes the internal method purer.
> * Have you checked to see if there is a test for the modified method?

Yes, right.

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 3]
```diff
@@ -387,37 +387,31 @@ protected void internalRevokePermissionsOnTopic(AsyncResponse asyncResponse, Str
...
-            Topic createdTopic = getOrCreateTopic(topicName, properties);
-            log.info("[{}] Successfully created non-partitioned topic {}", clientAppId(), createdTopic);
-        } catch (Exception e) {
-            if (e instanceof RestException) {
-                throw (RestException) e;
-            } else {
-                log.error("[{}] Failed to create non-partitioned topic {}", clientAppId(), topicName, e);
-                throw new RestException(e);
-            }
+            ret.thenCompose(__ -> validateGlobalNamespaceOwnershipAsync(namespaceName));
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:394]:
```suggestion
            ret = ret.thenCompose(__ -> validateGlobalNamespaceOwnershipAsync(namespaceName));
```

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: fixed.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**